src/cozmo/action.py

```diff
@@ -442,13 +448,22 @@ def _abort_action(self, action):
         # message back in the next engine tick, and can basically be considered
         # cancelled from now.
         action._set_aborting()
-        # move from in-progress to aborting dicts
-        self._aborting[action._action_id] = action
-        del self._in_progress[action._action_id]
 
-        msg = _clad_to_engine_iface.CancelActionByIdTag(idTag=action._action_id,
-                                                        robotID=self.robot.robot_id)
-        self.robot.conn.send_msg(msg)
+        if action._completed_event_pending:
+            # The action was marked as still running but the ActionDispatcher
+            # has already received a completion message (and removed it from
+            # _in_progress) - the action is just waiting to receive a
+            # robot_completed_action message that is still being dispatched
+            # via asyncio.ensure_future
+            logger.debug('Not sending abort for action=%s to engine as it just completed', action)
+        else:
+            # move from in-progress to aborting dicts
```

Comment:
The else case isn't new, it's just indented, but github's diff isn't making that clear - you can add `?w=1` on the end of github diff urls to ignore whitespace which is handy sometimes - e.g: https://github.com/anki/cozmo-python-sdk/pull/78/files/d534520b8a575284cc129f479148b5289655730a?w=1